modules/host/contractmanager/storagefoldershrink_bench_test.go

```diff
@@ -0,0 +1,38 @@
+package contractmanager
+
+import (
+	"testing"
+)
+
+// BenchmarkUnsetManySectors checks that unsetting a bunch of sectors
+// individually takes an acceptable amount of time. Millions should be possible
+// in well under a second.
+//
+// My laptop is doing 10e6 in 1.2 seconds. This is on the edge of too slow, but
+// overall it's close enough.
+func BenchmarkUnsetManySectors(b *testing.B) {
+	// Create a uint64 that has all of the bits set.
+	base := uint64(0)
+	base--
+
+	// Benchmark how long it takes to unset the bits.
+	for i := 0; i < b.N; i++ {
+		// Create a usage array with all bits set.
+		b.StopTimer()
+		usageArray := make([]uint64, 10e6)
+		for i := 0; i < len(usageArray); i++ {
+			usageArray[i] = base
+		}
+		b.StartTimer()
```

Comment:
I highly doubt that, but I suppose it's not impossible. By the same token, the compiler might optimize your bit-setting loop to a single `&`.
I have long been curious about how aggressively the compiler will optimize benchmarks. It seems like a Sufficiently Smart Compiler could easily see that `usageArray` isn't being used anywhere, and decide to remove all the code being benchmarked. I think this actually does happen for really trivial code. Nowadays I frequently use `_ = fn()` to indicate that the function call shouldn't be optimized out.

Anyway, you should at least move the allocation out of the loop. There's no reason to be allocating 10e6 bytes inside a tight benchmarking loop. It won't affect the timer, but it still makes the benchmark run more slowly.